pkg/config/expand.go

```diff
@@ -143,9 +144,24 @@ func (bp Blueprint) expandGroup(gp groupPath, g *Group) error {
 func (bp Blueprint) expandModule(mp ModulePath, m *Module) error {
 	bp.applyUseModules(m)
 	bp.applyGlobalVarsInModule(m)
+	// Versioned Module Logic for Embedded Modules
+	if bp.ToolkitModulesURL != "" && bp.ToolkitModulesVersion != "" {
```

Comment:
AFAIK, this is too late for substituting source, e.g.:
`validateModule` has already been executed by this point, validation will be done agains "non-doctored" source.